### ISSUE TITLE
MAINT:linalg: Either silent import NumPy versions or deprecate tril, triu, tri 

### ISSUE THREAD

**Issue by ilayn** [scipy/scipy#15702]
This is a reminder for myself which is a part of linalg import clean-up attempts. As Ralf found out all modules are now sphagettified over time importing each other. But also, there are some unnecessary imports across internal module files anyways.  

For some historical reason, these still live in `linalg._special_matrices.py` but identical to NumPy functionality. From the top of the file the notice says 

```
# *Note*: tri{,u,l} is implemented in NumPy, but an important bug was fixed in
# 2.0.0.dev-1af2f3, the following tri{,u,l} definitions are here for backwards
# compatibility
```

that points to https://github.com/numpy/numpy/commit/1af2f37 I think 1.7.0 is meant here anyways and after 10 years I think this is enough time (https://github.com/numpy/numpy/issues/2441). Plus these suffer from the bug that is fixed by that commit. 

As far as I can tell there is no extra functionality offered anyways and clutters the namespace which is already cluttering NumPy namespace. I just found this out when I was checking `linalg._matfuncs.py` with a single occurence.

I'd say let's get rid of them. Thoughts?

**Comment by ilayn** [comment]
Also it seems like we lost the commit history on the GitHub interface when we underscore'd files. Is there any way to fix that or it's a GH bug?

**Comment by h-vetinari** [comment]
`blame` still works: https://github.com/scipy/scipy/blame/main/scipy/linalg/_special_matrices.py

You can also go to [history](https://github.com/scipy/scipy/commits/main/scipy/linalg/_special_matrices.py) -> oldest [commit](https://github.com/scipy/scipy/commit/6d9ce3a972c65328e19a3624027f8a0af5163745) -> [parent](https://github.com/scipy/scipy/commit/ca5e4f012f2ff847ff64310d81a29d2d06e353f3) -> browse files -> go to [`special_matrices.py`](https://github.com/scipy/scipy/blob/ca5e4f012f2ff847ff64310d81a29d2d06e353f3/scipy/linalg/special_matrices.py) -> [history](https://github.com/scipy/scipy/commits/09a001e2bdfa3cf604dd5f15f2a20bd48274fa35/scipy/linalg/special_matrices.py).

**Comment by h-vetinari** [comment]
That said, it seems to be a (theoretically unnecessary) limitation of the GH interface that the history does not extend across renames.

**Comment by ilayn** [comment]
I'll check if there is an open ticket in the maintainer feedback group. Otherwise probably someone should have opened an issue about it.

**Comment by AnirudhDagar** [comment]
@ilayn there is an easier way to follow history now after renames. Recently shipped in June: https://github.blog/changelog/2022-06-06-view-commit-history-across-file-renames-and-moves/

**Comment by rgommers** [comment]
> I'd say let's get rid of them. Thoughts?

Yes, this seems like a good idea, +1 for deprecation.

**Comment by rgommers** [comment]
This can be done any time, so removed the milestone. Seems like a nice cleanup. @j-bowhay maybe you are interested in doing this?

**Comment by ilayn** [comment]
Fixed in #18021
